extension/content/components/recipes/RecipeEditor.js

```diff
@@ -0,0 +1,193 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import {
+  Alert,
+  Button,
+  ButtonToolbar,
+  Form,
+  FormGroup,
+  FormControl,
+  ControlLabel,
+  HelpBlock,
+  InputPicker,
+} from "rsuite";
+import { Controlled as CodeMirror } from "react-codemirror2";
+import {
+  useSelectedEnvironmentAPI,
+  useSelectedEnvironmentAuth,
+} from "devtools/contexts/environment";
+
+export default function RecipeEditor(props) {
+  const { match } = props;
+  const api = useSelectedEnvironmentAPI();
+  const auth = useSelectedEnvironmentAuth();
+  let [data, setData] = useState({});
+  let [actions, setActions] = useState([]);
+
+  async function getActionsOptions() {
+    let res = await api.fetchActions(3);
+    let actions = res.results.map(action => ({
+      label: action.name,
+      value: action.id,
+    }));
+
+    setActions(actions);
+  }
+
+  async function getRecipe(id) {
+    let recipe = await api.fetchRecipe(id, 3);
+    const { latest_revision } = recipe;
+    latest_revision.arguments = JSON.stringify(
+      latest_revision.arguments,
+      null,
+      1,
+    );
+    latest_revision.action_id = latest_revision.action.id;
+    delete latest_revision.comment;
+    setData(latest_revision);
```

Comment:
i think this should just be:
```js
const recipe = await api.fetchRecipe(id);
setData(recipe.latest_revision)
```

any manipulation of data should happen at the field level, not here

also you don't need the version passed through on `fetchRecipe` since it is using v3.